Techion.rte/Devices/Shared/Scripts/PulseShot.lua

```diff
@@ -20,37 +20,42 @@ function Update(self)
 			self.trailPar.Pos = self.Pos - Vector(self.Vel.X, self.Vel.Y):SetMagnitude(6);
 			self.trailPar.Vel = self.Vel * 0.5;
 			self.trailPar.Lifetime = self.Age + TimerMan.DeltaTimeMS;
+		else
+			self.trailPar = nil;
 		end
 	end
 	self.EffectRotAngle = self.Vel.AbsRadAngle;
 end
 function PulsarDissipate(self, inverted)
+	self.lastVel = self.lastVel or Vector(self.Vel.X, self.Vel.Y);
 
 	local trace = inverted and Vector(-self.Vel.X, -self.Vel.Y):SetMagnitude(GetPPM()) or Vector(self.Vel.X, self.Vel.Y):SetMagnitude(self.Vel.Magnitude * rte.PxTravelledPerFrame + 1);
 	local hit;
 	local hitPos = Vector();
 	local skipPx = math.sqrt(self.Vel.Magnitude) * 0.5;
 
-	local ray = SceneMan:CastObstacleRay(self.Pos, trace, hitPos, Vector(), self.ID, self.Team, rte.airID, skipPx);
-	if ray >= 0 then
-		local mo = MovableMan:GetMOFromID(SceneMan:GetMOIDPixel(hitPos.X, hitPos.Y));
+	local moid =	SceneMan:CastObstacleRay(self.Pos, trace, hitPos, Vector(), self.ID, self.Team, rte.airID, skipPx) >= 0
```

Comment:
Looks like some weird spacing going on here after the `=`

Reply:
It was to get the multiline thing lined up nicely... I'll just oneline it